flink-autoscaler/src/main/java/org/apache/flink/autoscaler/JobAutoScalerImpl.java

```diff
@@ -115,10 +123,39 @@ public void cleanup(KEY jobKey) {
         stateStore.removeInfoFromCache(jobKey);
     }
 
-    private void clearParallelismOverrides(Context ctx) throws Exception {
-        var parallelismOverrides = stateStore.getParallelismOverrides(ctx);
-        if (parallelismOverrides.isPresent()) {
+    private void clearStatesAfterAutoscalerDisabled(Context ctx) throws Exception {
+        var needFlush = false;
+        var parallelismOverridesOpt = stateStore.getParallelismOverrides(ctx);
+        if (parallelismOverridesOpt.isPresent()) {
+            needFlush = true;
             stateStore.removeParallelismOverrides(ctx);
+        }
+
+        Optional<SortedMap<Instant, CollectedMetrics>> collectedMetricsOpt =
+                stateStore.getCollectedMetrics(ctx);
+        if (collectedMetricsOpt.isPresent()) {
+            needFlush = true;
+            stateStore.removeCollectedMetrics(ctx);
+        }
+
+        Optional<Map<JobVertexID, SortedMap<Instant, ScalingSummary>>> scalingHistoryOpt =
+                stateStore.getScalingHistory(ctx);
+        if (scalingHistoryOpt.isPresent()) {
+            Map<JobVertexID, SortedMap<Instant, ScalingSummary>> scalingHistory =
+                    trimScalingHistory(
+                            clock.instant(), ctx.getConfiguration(), scalingHistoryOpt.get());
+            if (scalingHistory.isEmpty()) {
+                // All scaling histories are trimmed.
+                needFlush = true;
+                stateStore.removeScalingHistory(ctx);
+            } else if (!scalingHistoryOpt.get().equals(scalingHistory)) {
+                // Some scaling histories are trimmed.
+                needFlush = true;
+                stateStore.storeScalingHistory(ctx, scalingHistory);
+            }
+        }
```

Comment:
This looks strange. Why not simply call ` stateStore.removeScalingHistory(ctx);` all the time? Then we wouldn't need to touch the trimming logic.

Reply:
Trimming logic will consider `VERTEX_SCALING_HISTORY_AGE` option, and the default value is `24 hours`.

My thought is: if users enable autoscaler within 24 hours, the `ScalingHistory` can be used for subsequent scaling. WDYT?

Reply:
I see, makes sense